src/matrix/sync.rs

```diff
@@ -53,18 +54,21 @@ impl MatrixSyncClient {
             access_token,
             next_token: None,
             http_client: http::ClientWrapper::new(),
-            current_sync: RequestStatus::NoRequest,
+            current_sync: Mutex::new(RequestStatus::NoRequest),
             ctx,
         }
     }
 
     pub fn poll_sync(&mut self, cx: &mut Context<'_>) -> Poll<Result<SyncResponse, io::Error>> {
         task_trace!(self.ctx, "Polled sync");
         loop {
-            match &mut self.current_sync {
+            let mut current_sync = self.current_sync.lock().unwrap();
+
+            match &mut current_sync.deref_mut() {
```

Comment:
`Deref`s methods should never have to be used directly

```suggestion
            match &mut *current_sync {
```

Reply:
fixed